src/Hazelcast.Net/Partitioning/Partitioner.cs

```diff
@@ -127,13 +128,13 @@ public void SetOrVerifyPartitionCount(int count)
         {
             lock (_partitionsLock)
             {
-                if (_partitions == null)
+                if (_partitions == null && Count == 0)
                 {
                     Count = count;
                 }
                 else if (Count != count)
                 {
-                    throw new InvalidOperationException($"Received count value {count} but expected {Count}.");
+                    throw new ClientNotAllowedInClusterException($"Received partition count value {count} but expected {Count}.");
```

Comment:
Yeah, this would be more clean and separated. Returning boolean fits better to the separation. Fixed at https://github.com/hazelcast/hazelcast-csharp-client/pull/587/commits/66b28e3e6d08210f07fe686b7eee389cfdcc3665